### ISSUE TITLE
feature request: flag duplicated class properties

### ISSUE THREAD

**Issue by ghsanti** [astral-sh/ruff#11925]
I don't think there is a rule flagging this common mistake?

```python
class ThisOne:
    def __init__(self):
      self.hello="hi"
      self.hello="bye"   # <-- flag repeated property
```

**Comment by MichaReiser** [comment]
@AlexWaygood how common is it to reassign self variables in the constructor? Like, is this something that's frowned upon

```python
class Test:
	def __init__(self, arg):
		self.name = "hi"
		if arg:
			self.name += " more"
```

Either way, I think Ruff needs a control flow sensitive analysis to avoid too many false positives, or it would flag:

```python
class Test:
	def __init__(self, arg):
		if arg:
			self.message = "hy"
		else:
			self.message = "bye"
		
```

which should remain valid.

**Comment by AlexWaygood** [comment]
> @AlexWaygood how common is it to reassign self variables in the constructor? Like, is this something that's frowned upon

It's pretty common. I wouldn't say that that's frowned upon. If we were to accept this, we'd need a way to distinguish between "pointless" assignments that are immediately overriden, like OP's example, and assignments like your example there which seem fine to me.

**Comment by aka-santi** [comment]
Also is it possible to specify:

```
class ThisOne:
    hello: int 
   def __init__(self):
      self.hello="hi" # error, not an int
```

?
